packages/collections/list.pony

```diff
@@ -1,6 +1,96 @@
+  linked list can be conceptualized as two singly linked lists formed from the same data items, but
+  in opposite sequential orders.
+
+  As you would expect. functions are provided to perform all the common list operations such as
+  creation, traversal, node addition and removal, iteration, mapping, filtering, etc.
+  
+  ## Example program
+  There are a _lot_ of functions in List. The following code picks out a few common examples.
+
+  It outputs:
+
+      A new empty list has 0 nodes.
+      Adding one node to our empty list means it now has a size of 1.
+      The first (index 0) node has the value: A single String
+      A list created by appending our second single-node list onto our first has size: 2
+      The List nodes of our first list are now:
+        A single String
+        Another String
+      Append *moves* the nodes from the second list so that now has 0 nodes.
+      A list created from an array of three strings has size: 3
+        First
+        Second
+        Third
+      Mapping over our three-node list produces a new list of size: 3
+      Each node-value in the resulting list is now far more exciting:
+        First BOOM!
+        Second BOOM!
+        Third BOOM!
+  
+  ```pony
+    use "collections"
+
+    actor Main
+      new create(env:Env) =>
+        
+        // Create a new empty List of type String
+        let my_list = List[String]()
+        
+        env.out.print("A new empty list has " + my_list.size().string() + " nodes.") // 0
+        
+        // Push a String literal onto our empty List
+        my_list.push("A single String")
+        env.out.print("Adding one node to our empty list means it now has a size of "
+                      + my_list.size().string() + ".") // 1
+        
+        // Get the first element of our List
+        try env.out.print("The first (index 0) node has the value: "
+                          + my_list.index(0)?()?.string()) end // A single String
+        
+        // Create a second List from a single String literal
+        let my_second_list = List[String].unit("Another String")
+        
+        // Append the second List to the first
+        my_list.append_list(my_second_list)
+        env.out.print("A list created by appending our second single-node list onto our first has size: "
+                      + my_list.size().string()) // 2
+        env.out.print("The List nodes of our first list are now:")
+        for n in my_list.values() do
+          env.out.print("\t" + n.string())
+        end
+        // NOTE: this _moves_ the elements so second_list consequently ends up empty
+        env.out.print("Append *moves* the nodes from the second list so that now has "
+                      + my_second_list.size().string() + " nodes.") // 0
+        
+        // Create a third List from a Seq(ence)
+        // (In this case a literal array of Strings)
+        let my_third_list = List[String].from([ "First"; "Second"; "Third" ])
```

Comment:
Nit: [the style guide](https://github.com/ponylang/ponyc/blob/master/STYLE_GUIDE.md#array-literals) wants array literals to be like `["First"; "Second"; "Third"]`

Reply:
Thanks @srenatus ! I'll fix that right now.